crates/bevy_ecs/src/reflect.rs

```diff
@@ -19,18 +33,29 @@ pub struct ReflectComponent {
 }
 
 impl ReflectComponent {
-    pub fn add_component(&self, world: &mut World, entity: Entity, component: &dyn Reflect) {
-        (self.add_component)(world, entity, component);
+    /// Inserts the non-erased value of `component` (with type `C`) into the `entity`
+    ///
+    /// PANICS: `component` must have the same type `C` as the type used to create this struct
+    pub fn insert_component(&self, world: &mut World, entity: Entity, component: &dyn Reflect) {
+        (self.insert_component)(world, entity, component);
     }
 
+    /// Sets the existing value of type `C` found on `entity` to the non-erased value of `component`
+    ///
+    /// PANICS: `component` must have the same type `C` as the type used to create this struct
```

Comment:
Use the conventional `# Panics` markup here.

Reply:
```suggestion
    /// # Panics
    /// `component` must have the same type `C` as the type used to create this struct
```